seq2science/rules/configuration_workflows.smk

```diff
@@ -153,47 +152,3 @@ if config.get("contrasts"):
             "A DESeq2 design contrast must be in the form '(batch+)column_target_reference'. See the docs for examples.\n"
         )
         _, _, _, _ = parse_contrast(contrast, samples, check=True)
-
-
-def get_read_length(sample):
-    """
-    This function returns the read length for a sample.
-
-    Depending on the fastq qc tool it raises a checkpoint exception, waits for that
-    checkpoint to have been executed, and then checks in the qc summary of the sample
-    what the (average) read length is after trimming.
-    """
-    # if dryrunning just pretend we know already
-    if workflow.persistence.dag.dryrun:
-        return 40
-
-    # trimgalore
-    if config["trimmer"] == "trimgalore":
-        if sampledict[sample]["layout"] == "SINGLE":
-            qc_file = checkpoints.fastqc.get(fname=f"{sample}_R1_trimmed").output.zip
-            zip_name = f"{sample}_trimmed_fastqc/fastqc_data.txt"
-        if sampledict[sample]["layout"] == "PAIRED":
-            qc_file = checkpoints.fastqc.get(fname=f"{sample}_R1_trimmed").output.zip
-            zip_name = f"{sample}_{config['fqext1']}_trimmed_fastqc/fastqc_data.txt"
-
-        import zipfile
-        import re
-
-        qc_report = zipfile.ZipFile(qc_file, "r").read(zip_name)
-
-        kmer_size = re.search("Sequence length\\t(\d+)", qc_report.decode("utf-8")).group(1)
-
-    # fastp
-    if config["trimmer"] == "fastp":
-        if sampledict[sample]["layout"] == "SINGLE":
-            qc_file = checkpoints.fastp_SE.get(sample=sample).output.qc_json[0]
-        if sampledict[sample]["layout"] == "PAIRED":
-            qc_file = checkpoints.fastp_PE.get(sample=sample).output.qc_json[0]
```

Comment:
It hurts to see you delete all your hard work :<

Suggestion: 
- turn this into a rule that outputs a temp file for each sample. 
- `macs2_callpeak` accepts this temp file as input, instead of the kmer size-specific file.
- the rule body executes the shell script in `get_effective_genome_size`, creating the kmer size-specific output, if it does not yet exist.
- the kmer size-specific file is copied and renamed to the temp output file.

that should remove the need for checkpoints, but now only 1 of these can run simultaneously to prevent double kmer file generation (and old/corrupted kmer files are never overwritten).

Reply:
I changed it so that there is now a genome size per sample, not per kmer size. Might decrease the down time a little bit.. :shrug: